test/unit/lodash.spec.js

```diff
@@ -0,0 +1,638 @@
+            });
+
+            it('should return total', function ()
+            {
+                expect(_.add.apply(_, datasets.add(params))).to.eql(-2);
+            });
+        });
+
+        describe('ceil', function ()
+        {
+            var params, number, precision;
+
+            before(function ()
+            {
+                params = datasets.ceil();
+                number = params[0];
+                precision = params[1];
+            });
+
+            it('should verify params', function ()
+            {
+                expect(params).to.have.length(2);
+                expect(typeof number).to.eql('number');
+                expect(typeof precision).to.eql('number');
+
+                expect(Number.isInteger(number)).to.not.eql(true);
+                expect(precision).to.not.eql(0);
+                expect(number).to.not.eql(1.85);
+            });
+
+            it('should return round up number', function ()
+            {
+                expect(_.ceil.apply(_, datasets.ceil(params))).to.eql(1.85);
+            });
+        });
+
+        describe('divide', function ()
+        {
+            var params, dividend, divisor;
+
+            before(function ()
+            {
+                params = datasets.divide();
+                dividend = params[0];
+                divisor = params[1];
+            });
+
+            it('should verify params', function ()
+            {
+                expect(typeof dividend).to.eql('number');
+                expect(typeof divisor).to.eql('number');
+                expect(dividend).to.not.eql(Infinity);
+
+                expect(params).to.have.length(2);
+            });
+
+            it('should divide two numbers', function ()
+            {
+                expect(_.divide.apply(_, datasets.divide(params))).to.eql(Infinity);
+            });
+
+        });
+
+        describe('floor', function ()
+        {
+            var params, number, precision;
+
+            before(function ()
+            {
+                params = datasets.floor();
+                number = params[0];
+                precision = params[1];
+            });
+
+            it('should verify params', function ()
+            {
+                expect(params).to.have.length(2);
+                expect(typeof number).to.eql('number');
+                expect(typeof precision).to.eql('number');
+                expect(number).to.not.eql(5.47);
+
+                expect(Number.isInteger(number)).to.not.eql(true);
+            });
+
+            it('should return number round down', function ()
+            {
+                expect(_.floor.apply(_, datasets.floor(params))).to.eql(5.47);
+            });
+        });
+
+        describe('max', function ()
+        {
+            var params, array;
+
+            before(function ()
+            {
+                params = datasets.max();
+                array = params[0];
+                Array.prototype.checkIfArrayIsUnique = function ()
```

Comment:
`checkIfArrayIsUnique` prototype is used in many describes  - we should move definition to one `before {}` for example here 

```
before(function(){
Array.prototype.checkIfArrayIsUnique  = ...

});

describe('Math functions', function ()
{
        describe('...', function(){}...

});

```